Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask_test.py

```diff
@@ -0,0 +1,143 @@
+import pytest
+
+from WaitAndCompleteTask import wait_and_complete_task_command
+
+TASK_BY_STATES = [
+    {
+        'name': 'conditional task 1',
+        'id': 1,
+        'parentPlaybookID': 123,
+        'state': 'Waiting'
+    },
+    {
+        'name': 'conditional task 2',
+        'id': 2,
+        'parentPlaybookID': 123,
+        'state': 'Waiting'
+    },
+    {
+        'name': 'manual task 1',
+        'id': 3,
+        'parentPlaybookID': 123,
+        'state': 'Waiting'
+    },
+    {
+        'name': 'manual task 2',
+        'id': 4,
+        'parentPlaybookID': 123,
+        'state': 'Waiting'
+    }
+]
+
+COMPLETE_RES_ALREADY_COMPLETED = 'Task is completed already'
+COMPLETE_RES = '{"id": some_id}'
+
+
+@pytest.mark.parametrize('args,tasks_ret_value,complete_res,completed_tasks,found_tasks', [
+    ({'incident_id': 1,
+      'task_states': 'Waiting',
+      'complete_option': 'yes',
+      'task_name': 'conditional task 1',
+      'complete_task': 'true',
+      'max_iterations': '1'
+      },
+     TASK_BY_STATES, COMPLETE_RES, ['conditional task 1'], []),
+    ({'incident_id': 1,
+      'task_states': 'Waiting',
+      'task_name': 'conditional task 1',
+      'max_iterations': '1',
+      'complete_task': 'false',
+      },
+     TASK_BY_STATES, COMPLETE_RES, [], ['conditional task 1']),
+    ({'incident_id': 1,
+      'task_states': 'Waiting',
+      'max_iterations': '1',
+      'complete_task': 'true',
+      },
+     TASK_BY_STATES, COMPLETE_RES, ['conditional task 1', 'conditional task 2', 'manual task 1', 'manual task 2'],
+     []),
+    ({'incident_id': 1,
+      'task_states': 'Waiting',
+      'max_iterations': '1',
+      'complete_task': 'true',
+      },
+     TASK_BY_STATES, COMPLETE_RES_ALREADY_COMPLETED, [],
+     ['conditional task 1', 'conditional task 2', 'manual task 1', 'manual task 2']),
+    ({'incident_id': 1,
+      'task_states': 'Waiting',
+      'max_iterations': '1',
+      'complete_task': 'false',
+      },
+     TASK_BY_STATES, '', [],
+     ['conditional task 1', 'conditional task 2', 'manual task 1', 'manual task 2'])
+])
+def test_wait_and_complete_task_command(mocker, args, tasks_ret_value, complete_res, completed_tasks, found_tasks):
+    """
+    Given:
+        - The "WaitAndCompleteTask" script
+
+    When:
+        - Waiting to all task that has given status
```

Comment:
```suggestion
        - An incident which runs a playbook with four tasks in a Waiting state.
    When:
        - Calling the "WaitAndCompleteTask" script in the following cases:
            - ...
```
in the "when" and "then" sections please describe the cases.